mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -166,8 +167,8 @@ def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
                 prev_val_length = len(str(val))
             elif len(str(val)) != prev_val_length:
                 is_same_length = False
-            else:
-                prev_val_length = len(str(val))
+
+            prev_val_length = len(str(val))
 
```

Comment:
The for loops can be simplified to this:
```
data = [str(x) for x in data]
all_same_length = all(len(data[0]) == len(x) for x in data)
uuid_charset = set('0123456789abcdef-')
is_uuid = all(set(x).issubset(uuid_charset) for x in data)
data_looks_like_id = (all_same_length and is_uuid)
```

Reply:
made some changes based on this.